### PR TITLE
feat(server): adds S3 client to upload images with new bike endpoint

### PR DESCRIPTION
**Pull request by Mackbellemore**
#

## Changes this PR introduces
first image attached to a bike https://soen390.s3-us-west-2.amazonaws.com/603726757a92b8001347072b.jpg 
:tada: 
- new PATCH /bikes/image/:id which takes formdata file: image
- Uses localstack to mock aws resources locally
- this PR does not affect anything, besides the csv download but I default the img url to null for now so we good

## How to test this PR

- PATCH /bikes/image/:id with formdata named file and with jpeg, jpg or png
image

## Notes for reviewers

- a note

## Checklist

- [x] I renamed the title of my pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-rqaw9j Allow image uploads for Bikes](https://app.clickup.com/t/rqaw9j)

File: server/app/services/S3Service.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  public contructUrl(key: string): string {
+    const isDev = this.config.get<string>('env') === 'development';
+    const bucket = this.config.get<string>('aws.bucket');
+    let url = '';
+    if (isDev) {
+      url = `http://localhost:4566/${bucket}/${key}`;
+    } else {
+      url = `https://${bucket}.s3-${this.config.get<string>('aws.region')}.amazonaws.com/${key}`;
+    }
```
Comment (gordonpn) [inline server/app/services/S3Service.ts]:
```suggestion
    let url = `https://${bucket}.s3-${this.config.get<string>('aws.region')}.amazonaws.com/${key}`;
    if (isDev) {
      url = `http://localhost:4566/${bucket}/${key}`;
    }
```

😀

Comment (Mackbellemore) [reply server/app/services/S3Service.ts]: wah

**Review by gordonpn — APPROVED** [review]
☁️ ☁️ ☁️

**Review by nasjuice — APPROVED** [review]
my man

**Pull request merged**